src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/custom.py

```diff
@@ -90,11 +114,233 @@ def _upload_package_blob(package_file, url):
+def create_application_package(client, resource_group_name, account_name,  # pylint:disable=too-many-arguments
                                application_id, version, package_file):
+
+    # create application if not exist
+    mgmt_client = get_mgmt_service_client(BatchManagementClient)
+    try:
+        mgmt_client.application.get(resource_group_name, account_name, application_id)
+    except:  # pylint:disable=W0702
+        mgmt_client.application.create(resource_group_name, account_name, application_id)
+
     result = client.create(resource_group_name, account_name, application_id, version)
+
+    # upload binary as application package
     logger.info('Uploading %s to storage blob %s...', package_file, result.storage_url)
     _upload_package_blob(package_file, result.storage_url)
-    return result
 
-create_application_package.__doc__ = ApplicationPackageOperations.create.__doc__
+    # activate the application package
+    client.activate(resource_group_name, account_name, application_id, version, "zip")
+    return client.get(resource_group_name, account_name, application_id, version)
+
+
+# Data plane custom commands
+
+@transfer_doc(CertificateAddParameter)
+def create_certificate(client, cert_file, thumbprint, thumbprint_algorithm, password=None):
+    if password:
+        certificate_format = 'pfx'
+    else:
+        certificate_format = 'cer'
+    with open(cert_file, "rb") as f:
+        data_bytes = f.read()
+    data = base64.b64encode(data_bytes).decode('utf-8')
+    cert = CertificateAddParameter(thumbprint, thumbprint_algorithm, data,
+                                   certificate_format=certificate_format,
+                                   password=password)
+    try:
+        client.add(cert)
+        return client.get(thumbprint_algorithm, thumbprint)
+    except BatchErrorException as ex:
+        try:
+            message = ex.error.message.value
+            if ex.error.values:
+                for detail in ex.error.values:
+                    message += "\n{}: {}".format(detail.key, detail.value)
+            raise CLIError(message)
+        except AttributeError:
+            raise CLIError(ex)
+    except (ValidationError, ClientRequestError) as ex:
+        raise CLIError(ex)
+
+
+def delete_certificate(client, thumbprint, thumbprint_algorithm, abort=None):
+    try:
+        if abort:
+            client.cancel_deletion(thumbprint_algorithm, thumbprint)
+        else:
+            client.delete(thumbprint_algorithm, thumbprint)
+    except BatchErrorException as ex:
+        try:
+            message = ex.error.message.value
+            if ex.error.values:
+                for detail in ex.error.values:
+                    message += "\n{}: {}".format(detail.key, detail.value)
+            raise CLIError(message)
+        except AttributeError:
+            raise CLIError(ex)
+    except (ValidationError, ClientRequestError) as ex:
+        raise CLIError(ex)
```

Comment:
This try/except logic is used multiple times.
Consider moving out to helper function that takes a callable.
e.g.:
```
def delete_certificate(...):
    def handler():
        if abort:
             client.cancel_deletion(thumbprint_algorithm, thumbprint)
         else:
             client.delete(thumbprint_algorithm, thumbprint)
    _ex_handle_patch_exceptions(handler)

def _ex_handle_patch_exceptions(some_handler):
    try:
        some_handler()
    except BatchErrorException ...
       ...
```

Something like that.